### PR TITLE
[Method Reference 3]LPS-173756 Replace method reference with hard reference in class SoapExtender

### PR DESCRIPTION
**Pull request by W2jx**
Hi @dantewang , I have repalced method reference with hard reference in class `SoapExtender`, but I keep the logic of manually registering the component, because this componet nedds to hard reference some `Bus` components according to the dynamically changing filter conditions. I don't see how this can be implemented in a standalone component, rather than relying on manually registering it outside the component.

The code: https://github.com/liferay/liferay-portal/blob/032c27b5c6c6849682394873d67a9e274bd5e26e/modules/apps/portal-remote/portal-remote-soap-extender-impl/src/main/java/com/liferay/portal/remote/soap/extender/internal/SoapExtender.java#L100C44-L107

### CODE REVIEW

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

File: modules/apps/portal-remote/portal-remote-soap-extender-api/src/main/java/com/liferay/portal/remote/soap/extender/SoapDescriptorBuilder.java [hunk 1]
```diff
@@ -18,9 +18,12 @@
 
 import javax.xml.namespace.QName;
 
+import org.osgi.annotation.versioning.ProviderType;
+
 /**
  * @author Carlos Sierra Andrés
  */
+@ProviderType
```
Comment (dantewang) [inline modules/apps/portal-remote/portal-remote-soap-extender-api/src/main/java/com/liferay/portal/remote/soap/extender/SoapDescriptorBuilder.java:26]: @W2jx Do not remove `@ProviderType`. This actually means the implementation/service should be provided by Liferay.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: readme/BREAKING_CHANGES.markdown [hunk 2]
```diff
@@ -1470,4 +1470,26 @@ This removal of the configuration has no replacement.
 
 ### Why was this change made?
 
-Liferay decided to not support this configuration.
\ No newline at end of file
+Liferay decided to not support this configuration.
```
Comment (dantewang) [inline readme/BREAKING_CHANGES.markdown:1473]: @tinatian Do we still need to record this breaking change if the interface `SoapDescriptorBuilder` is a provider type?

File: readme/BREAKING_CHANGES.markdown [hunk 3]
```diff
@@ -1470,4 +1470,26 @@ This removal of the configuration has no replacement.
...
+
+Custom `SoapDescriptor` is no longer supported.
+
+### Who is affected?
+
+This affects anyone implementing interfaces `SoapDescriptorBuilder` and `SoapDescriptor`.
+
+### How should I update my code?
+
+Refer to the properties defined in class `DefaultSoapDescriptor` to pass the corresponding `SOAP` data in a standardized manner.
```
Comment (dantewang) [inline readme/BREAKING_CHANGES.markdown:1491]: @W2jx Please provide a brief description about how to pass this data in a standard manner.

File: readme/BREAKING_CHANGES.markdown [hunk 4]
```diff
@@ -1470,4 +1470,26 @@ This removal of the configuration has no replacement.
...
+
+### Why was this change made?
+
+Liferay decided to not support this custom `SoapDescriptor`.
```
Comment (dantewang) [inline readme/BREAKING_CHANGES.markdown:1495]: @W2jx Please note that `SOAP` is deprecated in 7.3.

File: modules/apps/portal-remote/portal-remote-soap-extender-impl/src/main/java/com/liferay/portal/remote/soap/extender/internal/SoapExtender.java [hunk 5]
```diff
@@ -67,22 +65,6 @@ protected void deactivate() {
 		_dependencyManager.clear();
 	}
 
-	@Reference(
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY
-	)
-	protected void setSoapDescriptorBuilder(
```
Comment (dantewang) [inline modules/apps/portal-remote/portal-remote-soap-extender-impl/src/main/java/com/liferay/portal/remote/soap/extender/internal/SoapExtender.java:74]: @W2jx You didn't remove the `unsetSoapDescriptorBuilder`.

**Pull request closed without merging**